### ISSUE TITLE
Core/Gob: Collision is not update when gob state is changed.

### ISSUE THREAD

**Issue by Kittnz** [TrinityCore/TrinityCore#18045]
**Description**: The only problem remains with this gob is that collision is not accurate.
The collision used is from the default state and not updated when the state of the gob is changed.

**Current behaviour**: Gob keeps collision from the default state.
Checkout issue: https://github.com/TrinityCore/TrinityCore/issues/5754

**Expected behaviour**: It should update.

**Branch(es)**: 3.3.5 / 6.x (Select the branch(es) affected by this issue) 
Both

**TC hash/commit**:  
dac44a59b43e3857a074ea53cfb35d4ba6298bdb
**TDB version**:  
61

**Comment by Nawuko** [comment]
@Kittnz there are 2 object inside each other, the fountain itself has the normal "fountain" collision, and the "signs" on both sides add the collision of the "hero statue". So only spawn the fountain for servers without the achievment and spawn the fountain (active state) + the signs for servers where Lich King has fallen.

TL;DR: There is no problem with gameobjects, you misstaken by 2 objects inside eachother.

**Comment by Kittnz** [comment]
Then shouldn't this be handled by a script or something? If it is a fresh server (as trinitycore always is) only the fountain should be spawned and no the hero statue.

**Comment by Nawuko** [comment]
Yea there are a few ways, blizzard itself handles this with a dummy aura. It is casted by the floating "bunny" (trigger npc) above the fountain. I personally just have written a gameobject script (on the foutain) that checks every 5 minutes (and on the object creation) if the server first achievement is completed, if yes i just spawn the signs and toggle the fountain state, after that i disable the update time cause its no longer needed to check

**Comment by Aokromes** [comment]
blizzlike bug, i have mined on retail and after got minerals i was floating on void and on the altars of gundrak.

**Comment by Nawuko** [comment]
@Aokromes go home you are drunk!

**Comment by Kittnz** [comment]
Thats different @Aokromes  and that gets updated because if you move you fall down.

**Comment by Kittnz** [comment]
Checking on retail, for some reason they have the gob in the middle of dalaran has reverded to it's original state. And the collision is the same as on tc, so closing this.
